pkg/cmd/package/support/pack.go

```diff
@@ -0,0 +1,276 @@
+	"github.com/bmatcuk/doublestar/v4"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	FlagId        = "id"
+	FlagVersion   = "version"
+	FlagBasePath  = "base-path"
+	FlagOutFolder = "out-folder"
+	FlagInclude   = "include"
+	FlagVerbose   = "verbose"
+	FlagOverwrite = "overwrite"
+)
+
+type PackageCreateFlags struct {
+	Id        *flag.Flag[string]
+	Version   *flag.Flag[string]
+	BasePath  *flag.Flag[string]
+	OutFolder *flag.Flag[string]
+	Include   *flag.Flag[[]string]
+	Verbose   *flag.Flag[bool]
+	Overwrite *flag.Flag[bool]
+}
+
+type PackageCreateOptions struct {
+	*PackageCreateFlags
+	Writer   io.Writer
+	Ask      question.Asker
+	NoPrompt bool
+	CmdPath  string
+}
+
+func NewPackageCreateFlags() *PackageCreateFlags {
+	return &PackageCreateFlags{
+		Id:        flag.New[string](FlagId, false),
+		Version:   flag.New[string](FlagVersion, false),
+		BasePath:  flag.New[string](FlagBasePath, false),
+		OutFolder: flag.New[string](FlagOutFolder, false),
+		Include:   flag.New[[]string](FlagInclude, false),
+		Verbose:   flag.New[bool](FlagVerbose, false),
+		Overwrite: flag.New[bool](FlagOverwrite, false),
+	}
+}
+
+func NewPackageCreateOptions(f factory.Factory, flags *PackageCreateFlags, cmd *cobra.Command) *PackageCreateOptions {
+	return &PackageCreateOptions{
+		PackageCreateFlags: flags,
+		Writer:             cmd.OutOrStdout(),
+		Ask:                f.Ask,
+		NoPrompt:           !f.IsPromptEnabled(),
+		CmdPath:            cmd.CommandPath(),
+	}
+}
+
+func PackageCreatePromptMissing(opts *PackageCreateOptions) error {
+	if opts.Id.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Package ID",
+			Help:    "The ID of the package.",
+		}, &opts.Id.Value, survey.WithValidator(survey.ComposeValidators(
+			survey.Required,
+			survey.MaxLength(200),
+			survey.MinLength(1),
+		))); err != nil {
+			return err
+		}
+	}
+
+	if opts.Version.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Version",
+			Help:    "The version of the package; must be a valid SemVer; defaults to a timestamp-based version.",
+		}, &opts.Version.Value); err != nil {
+			return err
+		}
+	}
+
+	if opts.BasePath.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Base Path",
+			Help:    "Root folder containing the contents to zip; defaults to current directory.",
+		}, &opts.BasePath.Value); err != nil {
+			return err
+		}
+	}
+
+	if opts.OutFolder.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Out Folder",
+			Help:    "Folder into which the zip file will be written; defaults to the current directory",
+		}, &opts.OutFolder.Value); err != nil {
+			return err
+		}
+	}
+
+	if len(opts.Include.Value) == 0 {
+		for {
+			var pattern string
+			if err := opts.Ask(&survey.Input{
+				Message: "Include patterns",
+				Help:    "Add a file pattern to include, relative to the base path e.g. /bin/*.dll; defaults to \"**\".",
```

Comment:
could have the default value of `**\.` in the prompt
also the default could be moved to a const

Reply:
this one can not have a default in survey because of the loop it's in. If you don't provide a value its will set it to "**" making the loop never end, also makes it impossible to have "**" after providing other patterns.

Reply:
good point